skepticoin/scripts/mine.py

```diff
@@ -87,3 +95,30 @@ def main():
         print("Waiting for networking thread to stop")
         thread.join()
         print("Done; waiting for Python-exit")
+
+
+def main():
+    parser = DefaultArgumentParser()
+    parser.add_argument('-n', default=1, type=int, help='number of miner instances')
+    args = parser.parse_args()
+
+    wallet_lock = Lock()
+    if args.n > 1:
+        pids = [None] * args.n
+
+        for i in range(args.n):
+            if i > 0:
+                args.dont_listen = True
+            pids[i] = Process(target=miner, daemon=True, args=(args, wallet_lock))
+            pids[i].start()
+
+        try:
+            for pid in pids:
+                pid.join()
```

Comment:
They could be replaced with something like 
while True:
    sleep(0)
Which may be more clear. It just needs something with low cpu usage to keep the process busy while waiting for the KeyboardInterrupt or it will instantly get into the finally block and the interrupt will stop the joins and throw an error.